kgforge/specializations/models/rdf/service.py

```diff
@@ -272,3 +301,102 @@ def traverse_properties(properties) -> Tuple[Dict, Dict]:
+                    self._get_property_shapes_from_nodeshape(node_shape_uriref)
+                )
+                for triple_to_add in triples_to_add:
+                    schema_graph.add(triple_to_add)
+                for triple_to_remove in triples_to_remove:
+                    schema_graph.remove(triple_to_remove)
+            return schema_graph
+        else:
+            return self._graph.graph(URIRef(graph_id))
+
+    def _get_property_shapes_from_nodeshape(self, node_shape_uriref):
+        property_shapes_to_add = []
+        triples_to_add = []
+        triples_to_remove = []
+        schema_defines_shape = [
+            SH["and"] / (RDF.first | RDF.rest / RDF.first) * ZeroOrMore,
+            SH["or"] / (RDF.first | RDF.rest / RDF.first) * ZeroOrMore,
+            SH["xone"] / (RDF.first | RDF.rest / RDF.first) * ZeroOrMore,
+        ]
+        sh_nodes = []
+        sh_properties = []
+        sh_properties.extend(list(self._graph.objects(node_shape_uriref, SH.property)))
+        sh_nodes.extend(list(self._graph.objects(node_shape_uriref, SH.node)))
+        for s in schema_defines_shape:
+            sh_nodes.extend(list(self._graph.objects(node_shape_uriref, s / SH.node)))
+            sh_properties.extend(
+                list(self._graph.objects(node_shape_uriref, s / SH.property))
+            )
+        for sh_node in set(sh_nodes):
+            if (
+                str(sh_node) != str(node_shape_uriref)
+                and str(sh_node) != "http://www.w3.org/1999/02/22-rdf-syntax-ns#nil"
+            ):
+                t_a, p_a, t_r = self._get_property_shapes_from_nodeshape(sh_node)
+                if p_a:
+                    triples_to_add.extend(t_a)
+                    for propertyShape in p_a:
+                        triples_to_add.append(
+                            (node_shape_uriref, propertyShape[0], propertyShape[1])
+                        )
+                        property_shapes_to_add.append(propertyShape)
+                    triples_to_remove.extend(t_r)
+                    triples_to_remove.append((node_shape_uriref, SH.node, sh_node))
+
+        for sh_node_property in set(sh_properties):
+            if (
+                str(sh_node_property) != str(node_shape_uriref)
+                and str(sh_node_property)
+                != "http://www.w3.org/1999/02/22-rdf-syntax-ns#nil"
+            ):
+                property_shapes_to_add.append((SH.property, sh_node_property))
+
+        return triples_to_add, property_shapes_to_add, triples_to_remove
+
+    def get_shape_graph(self, shape_uriref: URIRef) -> Tuple[Shape, Graph]:
+        try:
+            shape = self._sg.lookup_shape_from_node(shape_uriref)
+            shape_graph = self.transitive_load_shape_graph(
+                self.shape_to_named_graph[shape_uriref],
+                self.shape_to_defining_resource[shape_uriref],
+                shape_uriref,
+            )
+        except KeyError:
```

Comment:
What's the reason for trying again the transitive_load_shape_graph? 
Is it transitive_load_shape_graph that can raise the KeyError or is it lookup_shape_from_node?